src/drawing.c

```diff
@@ -0,0 +1,139 @@
+// under the MIT license, by marius851000
+#include "drawing.h"
+#include "rng.h"
+
+const int VRAM_BASE = 0x6840000;
+
+struct DrawingState DRAWING_STATE;
+
+void init_drawing(const char* filepath) {
+    DRAWING_STATE.x = 256/2;
+    DRAWING_STATE.y = 192/2;
+    DRAWING_STATE.is_pen_down = 0;
+    DRAWING_STATE.progress_per_frame = 5;
+    DRAWING_STATE.buffer_pos = 0;
+    DRAWING_STATE.file_closed = 0;
+    DRAWING_STATE.color_base = 0x8000;
+    DataTransferInit();
+    FileOpen(&DRAWING_STATE.file, filepath);
```

Comment:
Need to call [`FileInit`](https://github.com/UsernameFodder/pmdsky-debug/blob/4f361d8590e4ba28306670a20046e1c6ba38fa59/symbols/arm9/libs.yml#L2098-L2108) or [`FileInitVeneer`](https://github.com/UsernameFodder/pmdsky-debug/blob/4f361d8590e4ba28306670a20046e1c6ba38fa59/symbols/arm9.yml#L932-L944) prior to file opening.